geometry/GoalManager.cpp

```diff
@@ -14,57 +36,36 @@ void GoalManager::SetGoals(std::map<int, Goal*> goals)
 
 void GoalManager::ProcessPedPosition(Pedestrian* ped)
 {
-//     Goal* goal = _allGoals[ped->GetFinalDestination()];
-//
-//     if (goal!=nullptr){
-//          if (WaitingArea* wa = dynamic_cast<WaitingArea*>(goal)) {
-//               bool pedInside = wa->IsInsideGoal(ped->GetPos());
-//
-//               if (pedInside) {
-//                    wa->addPed(ped->GetID());
-//
-//                    if (!wa->isOpen()) {
-//                         for (auto& goalItr : _allGoals) {
-//                              if (WaitingArea* goal = dynamic_cast<WaitingArea*>(goalItr.second)) {
-//                                   goal->updateProbabilities(false, wa->GetId());
-//                              }
-//                         }
-//                    }
-//
-//               }
-//               else {
-//
-//
-//                    // if just has left goal
-//                    //   decrease numPed in previous goal
-//                    //   if numPed <= maxPed
-//                    //        open Goal
-//                    //        update for goal where next goal has goal id
-//               }
-//          }
-//     }
-
      // Ped is in current waiting area
      if (CheckInsideWaitingArea(ped, ped->GetFinalDestination())){
           WaitingArea* wa = dynamic_cast<WaitingArea*>(_allGoals[ped->GetFinalDestination()]);
-          wa->addPed(ped->GetID());
+          // Add ped to waiting area
+          wa->AddPed(ped->GetID());
+          // Ped enters waiting area
           ped->EnterGoal();
-          if (!wa->isOpen()) {
+
+          // Check state of waiting area
+          if (!wa->IsOpen()) {
                SetState(wa->GetId(), false);
           }
      }
 
+     // Ped already had a goal AND has not reached final goal
```

Comment:
Instead of this comment you could rewrite the code to something more readable like:
```
const bool pedHasGoal = ped->GetLastGoalID() != -1 ;
const bool pedHasNotReachedFinalGoal = ped->GetFinalDestination() != ped->GetLastGoalID());
if(pedHasGoal && pedHasNotReachedFinalGoal) {
    // ... 
}